cli/build.rs

```diff
@@ -1,306 +1,410 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
 
-use deno_core::error::custom_error;
-use deno_core::error::AnyError;
-use deno_core::op;
-use deno_core::serde::Deserialize;
-use deno_core::serde_json::json;
-use deno_core::serde_json::Value;
-use deno_core::Extension;
-use deno_core::JsRuntime;
-use deno_core::OpState;
-use deno_core::RuntimeOptions;
-use regex::Regex;
-use std::collections::HashMap;
 use std::env;
 use std::path::Path;
 use std::path::PathBuf;
 
-// TODO(bartlomieju): this module contains a lot of duplicated
-// logic with `runtime/build.rs`, factor out to `deno_core`.
-fn create_snapshot(
-  mut js_runtime: JsRuntime,
-  snapshot_path: &Path,
-  files: Vec<PathBuf>,
-) {
-  // TODO(nayeemrmn): https://github.com/rust-lang/cargo/issues/3946 to get the
-  // workspace root.
-  let display_root = Path::new(env!("CARGO_MANIFEST_DIR")).parent().unwrap();
-  for file in files {
-    println!("cargo:rerun-if-changed={}", file.display());
-    let display_path = file.strip_prefix(display_root).unwrap();
-    let display_path_str = display_path.display().to_string();
-    js_runtime
-      .execute_script(
-        &("deno:".to_string() + &display_path_str.replace('\\', "/")),
-        &std::fs::read_to_string(&file).unwrap(),
-      )
-      .unwrap();
-  }
-
-  let snapshot = js_runtime.snapshot();
-  let snapshot_slice: &[u8] = &*snapshot;
-  println!("Snapshot size: {}", snapshot_slice.len());
-
-  let compressed_snapshot_with_size = {
-    let mut vec = vec![];
+use deno_runtime::*;
+
+mod ts {
+  use super::*;
+  use crate::deno_webgpu_get_declaration;
+  use deno_core::error::{custom_error, AnyError};
+  use deno_core::{op, Extension, JsRuntime, OpState, RuntimeOptions};
+  use regex::Regex;
+  use serde::Deserialize;
+  use serde_json::{json, Value};
+  use std::collections::HashMap;
+
+  // TODO(bartlomieju): this module contains a lot of duplicated
```

Comment:
Suggest to split this file into `build/mod.rs`, `build/tsc.rs` and `build/cli.rs`